demos/device_defender_for_aws/defender_demo.c

```diff
@@ -0,0 +1,649 @@
+            }
+        }
+        else
+        {
+            LogError( ( "Unexpected defender API : %d.", api ) );
+        }
+    }
+    else
+    {
+        LogError( ( "Unexpected publish message received. Topic: %.*s, Payload: %.*s.",
+                    ( int ) pPublishInfo->topicNameLength,
+                    ( const char * ) pPublishInfo->pTopicName,
+                    ( int ) pPublishInfo->payloadLength,
+                    ( const char * ) ( pPublishInfo->pPayload ) ) );
+    }
+}
+/*-----------------------------------------------------------*/
+
+static bool collectDeviceMetrics( void )
+{
+    int status = EXIT_FAILURE;
+    MetricsCollectorStatus_t metricsCollectorStatus;
+    uint32_t numOpenTcpPorts, numOpenUdpPorts, numEstablishedConnections;
+
+    /* Collect bytes and packets sent and received. */
+    metricsCollectorStatus = GetNetworkStats( &( networkStats ) );
+
+    if( metricsCollectorStatus != MetricsCollectorSuccess )
+    {
+        IotLogError( "GetNetworkStats failed. Status: %d.",
+                    metricsCollectorStatus );
+    }
+
+    /* Collect a list of open TCP ports. */
+    if( metricsCollectorStatus == MetricsCollectorSuccess )
+    {
+        metricsCollectorStatus = GetOpenTcpPorts( &( openTcpPorts[ 0 ] ),
+                                                  OPEN_TCP_PORTS_ARRAY_SIZE,
+                                                  &( numOpenTcpPorts ) );
+
+        if( metricsCollectorStatus != MetricsCollectorSuccess )
+        {
+            IotLogError( "GetOpenTcpPorts failed. Status: %d.",
+                        metricsCollectorStatus );
+        }
+    }
+
+    /* Collect a list of open UDP ports. */
+    if( metricsCollectorStatus == MetricsCollectorSuccess )
+    {
+        metricsCollectorStatus = GetOpenUdpPorts( &( openUdpPorts[ 0 ] ),
+                                                  OPEN_UDP_PORTS_ARRAY_SIZE,
+                                                  &( numOpenUdpPorts ) );
+
+        if( metricsCollectorStatus != MetricsCollectorSuccess )
+        {
+            LogError( ( "GetOpenUdpPorts failed. Status: %d.",
+                        metricsCollectorStatus ) );
+        }
+    }
+		
+    /* Collect a list of established connections. */
+    if( metricsCollectorStatus == MetricsCollectorSuccess )
+    {
+        metricsCollectorStatus = GetEstablishedConnections( &( establishedConnections[ 0 ] ),
+                                                            ESTABLISHED_CONNECTIONS_ARRAY_SIZE,
+                                                            &( numEstablishedConnections ) );
+
+        if( metricsCollectorStatus != MetricsCollectorSuccess )
+        {
+            IotLogError( "GetEstablishedConnections failed. Status: %d.",
+                        metricsCollectorStatus );
+        }
+    }
+
+    /* Populate device metrics. */
+    if( metricsCollectorStatus == MetricsCollectorSuccess )
+    {
+        status = EXIT_SUCCESS;
```

Comment:
```suggestion
        status = true;
```

Reply:
Made a commit to update and align the function commit type with demo helper.
e9d96e2